fiware-region-sanity-tests/commons/results_analyzer.py

```diff
@@ -97,52 +97,63 @@ def print_results(self):
 
-    def get_regions_with_key_test_cases_passed(self):
+    def print_global_status(self):
         """
-        This method will parse all test results for each Region and will take in account only test cases defined in the
-        settings.json file to consider is Regions is OK or NOT.
+        This method will parse test results for each Region and will take into account whether all key and/or optional
+        test cases are successful, according to the patterns defined in `settings.json`.
         :return:
         """
-
-        region_ok_list = []
-
         with open(PROPERTIES_FILE) as config_file:
             try:
                 conf = json.load(config_file)
             except Exception, e:
                 print 'Error parsing config file: %s' % e
 
-        key_test_cases_patterns = [re.compile(item) for item in conf[PROPERTIES_CONFIG_KEYTESTCASES]]
-        for item in self.dict:
-            passed = True
-            for result_value in self.dict[item]:
-                passed = True
-                for pattern in key_test_cases_patterns:
-                    if pattern.match(result_value['test_name']):
-                        if result_value['status'] != STATUS_OK:
-                            passed = False
-                            break
-                if not passed:
-                    break
-            if passed:
-                region_ok_list.append(item)
-
-        print "\nREGION GLOBAL STATUS\n"
-        print "Key test cases:", str(conf[PROPERTIES_CONFIG_KEYTESTCASES])
-        print "Region list with key test cases with PASSED status:"
-        if len(region_ok_list) == 0:
-            print "NONE!!!!!!!"
-        else:
-            for region in region_ok_list:
-                print " >> {}".format(region)
+        # dict holding global status according either key or optional test cases
+        global_status = {
+            GLOBAL_STATUS_OK: {
+                'caption': 'Regions satisfying all key test cases: %s' % conf[PROPERTIES_CONFIG_KEY_TEST_CASES],
+                'region_list': []
+            },
+            GLOBAL_STATUS_PARTIAL_OK: {
+                'caption': 'Regions only failing in optional test cases: %s' % conf[PROPERTIES_CONFIG_OPT_TEST_CASES],
+                'region_list': []
+            }
+        }
+
+        # check status
+        key_test_cases_patterns = [re.compile(item) for item in conf[PROPERTIES_CONFIG_KEY_TEST_CASES]]
+        opt_test_cases_patterns = [re.compile(item) for item in conf[PROPERTIES_CONFIG_OPT_TEST_CASES]]
+        for region, results in self.dict.iteritems():
+            key_test_cases = [
+                item for item in results
+                if any(pattern.match(item['test_name']) for pattern in key_test_cases_patterns)
+            ]
+            non_opt_test_cases = [
+                item for item in results
+                if all(not pattern.match(item['test_name']) for pattern in opt_test_cases_patterns)
```

Comment:
I think the result in *non_optional_test_cases* list should not consider ALL (key + non key) test cases that do NOT match with the *opt_test_cases* patterns, but only key test cases: Non key test cases are optionals by definition.
I mean: this logic should be applied in items of the key_test_cases results.